whd/dispatcher.py

```diff
@@ -137,15 +138,20 @@ def _trigger_resource_check(self, concourse_api, resources):
                 continue
             resources.insert(0, resource)
 
+        executor = ThreadPoolExecutor()
         for resource in resources:
-            app.logger.info('triggering resource check for: ' + resource.name)
```

Comment:
through parallelisation, you break the guaranteed execution order (see comment a few lines above)